### PR TITLE
fix(apex): pin da3 metric model in matrix runner

### PR DESCRIPTION
**Pull request by RC219805**
## Summary
- Scheduled APEX real runs with backend_id=da3 were constructing EnhanceConfig without an explicit model_key, allowing the legacy ModelVariant.METRIC_LARGE selector to resolve to the research DA3 model.
- Pin the runner's da3 config to da3-metric and cover real-run config construction with a focused regression test.

## Changes
- Added an APEX_DA3_MODEL_KEY constant and backend-scoped selector helper in scripts/apex_matrix_runner.py.
- Plumbed model_key into EnhanceConfig for backend_id=da3 while leaving non-da3 backends unset.
- Added tests/test_apex_matrix_runner.py to assert real da3 runs pass model_key=da3-metric before orchestration.
- No route, API, selector, CLI flag, or schema contract changes.

## Validation
Proven green:
- .venv/bin/pytest tests/test_apex_matrix_runner.py tests/test_apex_backend_deps.py tests/test_apex_contract_verification.py::TestExecutionModeEnforcement -v
- Commit hooks: Auto-format staged Python files, Black, isort, script topology, ML isolation, unsafe torch.load, gitleaks, and other configured pre-commit checks

Still failing:
- None observed in the narrow local gate.

Failure classification:
- No product, stale-test, or environment/tooling failures observed.

## Risk
- Low operational risk: the change is scoped to APEX matrix runner config construction for backend_id=da3.
- Revert-safe: removing the helper and model_key assignment restores prior behavior; no migrations or public contracts changed.

### CODE REVIEW

File: scripts/apex_matrix_runner.py [hunk 1]
```diff
@@ -229,6 +231,13 @@ def check_ml_dependencies(backend_id: str) -> tuple[bool, list[str]]:
     return all_available, missing
 
 
+def _model_key_for_backend(backend_id: str) -> Optional[str]:
+    """Return governed model selectors that the matrix runner must pin."""
+    if backend_id == "da3":
+        return APEX_DA3_MODEL_KEY
```
[the remark this answers is not in the document]
Comment (RC219805) [reply scripts/apex_matrix_runner.py]: Fixed in 80c0157be. The matrix runner now normalizes backend_id with normalize_backend_id(..., warn=False) before choosing the governed DA3 model key, so da3 plus depth-anything-v3 and depth_anything_v3 all pin da3-metric. I also updated the regression test to cover those aliases and validated the lint/import-boundary failure class locally: .venv/bin/pytest tests/test_apex_matrix_runner.py tests/test_apex_backend_deps.py tests/test_apex_contract_verification.py::TestExecutionModeEnforcement tests/structural/test_structural_integrity.py::test_import_boundaries_are_respected -v and PYTHON_BIN=.venv-lint/bin/python LINT_RUNNER_GITHUB_EVENT_NAME=pull_request ./scripts/lint_runner.sh pr.
